packages/vite/src/node/plugins/vendorAnalysis.ts

```diff
@@ -0,0 +1,244 @@
+          }
+        })
+      )
+      isDebug && debug(`${timeFrom(collectRedirectsStart)} collect redirects`)
+
+      /* trace all files imported by entry */
+      const traced = new Set<string>()
+      const traceImport = async (ids: Set<string>, imps: Set<string>) => {
+        const nextids = new Set<string>()
+        // push the import to vendor
+        const addImport = (imp: string) => {
+          // if it's a redirect, use the resolved id
+          const getTrueImp = (imp: string): string =>
+            redirects.has(imp) ? getTrueImp(redirects.get(imp)!) : imp
+          const trueImp = getTrueImp(imp)
+          const impId = trueImp.split('/+/', 2)[0]
+          imps.add(trueImp)
+          nextids.add(impId)
+        }
+        await Promise.all(
+          Array.from(ids).map(async (module) => {
+            if (traced.has(module)) return
+            traced.add(module)
+            const info = this.getModuleInfo(module)
+            if (!info || !info.code) return
+            // get all imports from module
+            const imports = await findImports(info)
+            for (const imp of imports) {
+              // ignore dynamic imports
+              if (imp.d !== -1 || !imp.n) continue
+              // resolve the absolute id
+              const id = (await this.resolve(imp.n, module))?.id
+              if (!id) continue
+              const statment: string = info.code.substring(imp.ss, imp.se)
+              if (
+                statment.startsWith('import "') ||
+                statment.startsWith("import '") ||
+                statment.startsWith('import *')
+              ) {
+                // import "foo.js"
+                // import * as foo from "foo.js"
+                nextids.add(id)
+              } else {
+                // import def, { foo, bar } from "foo.js"
+
+                // impDef: def
+                // imps:  foo, bar
+                const [, impDef, imps] =
+                  statment.match(
+                    /(?:import|export)\s+(?:([^,\s{]+)\s*)?(?:\s*,\s*)?(?:\{([^\}]+))?/
+                  ) || []
+
+                // default import
+                if (impDef) {
+                  addImport(id + '/+/default')
+                }
+                // named imports
+                if (imps) {
+                  for (const name of imps.split(',')) {
+                    const [, impName] =
+                      name.trim().match(/([\S]+)(?:\s+as\s+([\S]+))?/) || []
+                    if (!impName) continue
+                    addImport(id + '/+/' + impName)
+                  }
+                }
+              }
+            }
+          })
+        )
+        if (nextids.size) await traceImport(nextids, imps)
+      }
+      const vendorImps = new Set<string>()
+      const startTrace = performance.now()
+      await traceImport(entrieIds, vendorImps)
+      isDebug && debug(`${timeFrom(startTrace)} trace imports`)
+
+      /* collect all files imported by entry */
```

Comment:
should use one type of commenting for consistency
```suggestion
      // collect all files imported by entry
```